frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/common/useEditFieldForm.ts

```diff
@@ -51,25 +95,63 @@ export const useEditFieldForm = <FormShape, FieldShape extends FieldBase>({
 
   const watchedInputs = editForm.watch()
 
+  const onSaveSuccess = useCallback(
+    (newField) => {
+      editForm.reset(
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        transform.input(newField),
+      )
+      if (isMobile) {
+        onMobileModalClose()
+      }
+    },
+    [editForm, isMobile, onMobileModalClose, transform],
+  )
+
+  const handleUpdateField = editForm.handleSubmit((inputs) => {
+    const updatedFormField = transform.output(inputs, field)
+    if (stateData.state === BuildFieldState.CreatingField) {
+      return createFieldMutation.mutate(updatedFormField, {
+        onSuccess: onSaveSuccess,
+      })
+    } else if (stateData.state === BuildFieldState.EditingField) {
+      return editFieldMutation.mutate(
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```

Comment:
Casting wouldn't work here like it's done below for `updateEditState()`?

Reply:
tried it and it worked, thanks! eb40295a